application/client.core/src/app/environment/components/sidebar/shell/information/component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnInit, OnDestroy, ChangeDetectorRef } from '@angular/core';
+
+import ElectronIpcService, { IPCMessages } from '../../../../services/service.electron.ipc';
+import ShellService from '../services/service';
+
+import * as Toolkit from 'chipmunk.client.toolkit';
+
+interface IInformation {
+    env: { [key: string]: string};
+    shells: string[];
+    shell: string;
+    pwd: string;
+}
+
+@Component({
+    selector: 'app-sidebar-app-shell-information',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less']
+})
+
+export class SidebarAppShellInformationComponent implements OnInit, OnDestroy {
+
+    public _ng_information: IInformation = {
+        env: {},
+        pwd: '',
+        shell: '',
+        shells: []
+    };
+
+    private _logger: Toolkit.Logger = new Toolkit.Logger('SidebarAppShellInformation');
+    private _destroyed = false;
+
+    constructor(private _cdRef: ChangeDetectorRef) {}
+
+    public ngOnInit() {
+        this._init();
+    }
```

Comment:
Because it's async operation, even it would take less than 1 second, it's logical to have here a spinner (while IPC request is processing) as feedback to the user.